### DISCUSSION TITLE
Do I need a Provider?

### DISCUSSION THREAD

**Discussion by imcvakt** [pmndrs/jotai#384] [Q&A] [answered]
Guys, I'm doing some examples which works even without a provider.

So...

- Do I need a provider? 

_Atom configs don't hold values. Atom values are stored in a Provider. A Provider can be used like React context provider. Usually, we place one Provider at the root of the app, however you could use multiple Providers, each storing different atom values for its component tree._. 
[Core API Docs](https://github.com/pmndrs/jotai/blob/master/docs/api/core.md)

What's the difference if I do not use a Provider?

**Comment by Aslemammad** [comment, marked as answer]
> Do I need a provider?

Hmm, yes and no, we introduced [provider-less mode](https://twitter.com/dai_shi/status/1364203403023314952) a time ago. If you have a SSR app that needs to provide initial values for atoms on the server-side render phase, yes, you have to use Provider with initialValue prop in it. But if not, you don't need to use the Provider anymore. Or, for better debugging and seeing all values, we use the provider. 
> Can I have multiple providers?

Yes, why not? You can use more providers to provide different values for different subtrees.
> What's the difference?

![image](https://user-images.githubusercontent.com/37929992/112631484-913ca580-8e54-11eb-9b50-04d2d7cd5755.png)

**Reply by dai-shi** [reply]
> You can use the scope prop to use more providers.

A minor correction: The scope is for a different purpose. Even without scope, you can use multiple providers to provide different values for different sub trees (even nested).

**Reply by dai-shi** [reply]
You need Provider to see all values in the provider for debugging purpose.

**Reply by Aslemammad** [reply]
Thanks, @dai-shi, Fixed the answer now.

**Reply by imcvakt** [reply]
@Aslemammad  Thanks for your help!
@dai-shi  what's the purpose of a scope so?

**Reply by dai-shi** [reply]
scope is for library use case. library authors can use jotai internally without interfering app code, even if the app uses jotai.
